cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -274,6 +278,399 @@ var _ = Describe("Operation Controller", func() {
 			deleteTestResources(ctx, dbQueries, resourcesToBeDeleted)
 
 		})
+
+		Context("Process Application Operation Test", func() {
+			It("Verify that When an Operation row points to an Application row that doesn't exist, any Argo Application CR that relates to that Application row should be removed.", func() {
+				By("Close database connection")
+				defer dbQueries.CloseDatabase()
+
+				applicationCR := &appv1.Application{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Application",
+						APIVersion: "argoproj.io/v1alpha1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: "my-user",
+						Labels: map[string]string{
+							dbID: "doesnt-exist",
+						},
+						DeletionTimestamp: &metav1.Time{
+							Time: time.Now(),
+						},
+					},
+				}
+
+				err = task.event.client.Create(ctx, applicationCR)
+				Expect(err).To(BeNil())
+
+				go func() {
+					err = wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+						if applicationCR.DeletionTimestamp != nil {
+							err = k8sClient.Get(ctx, client.ObjectKeyFromObject(applicationCR), applicationCR)
+							Expect(err).To(BeNil())
+
+							applicationCR.Finalizers = nil
+
+							err = k8sClient.Update(ctx, applicationCR)
+							Expect(err).To(BeNil())
+
+							err = task.event.client.Delete(ctx, applicationCR)
+							return true, nil
+						}
+						return false, nil
+					})
+				}()
+
+				clusterCredentials := db.ClusterCredentials{
+					Clustercredentials_cred_id: string(uuid.NewUUID()),
```

Comment:
Yes It is boilerplate @jgwest I have added this code because we need this **gitopsEngineCluster, _, err := dbutil.GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID(ctx, string(kubesystemNamespace.UID), dbQueries, logger)** to create GitopsEngineCluster to proceed further with test cases, If I use  **db. CreateSampleData()**  It will create 2 GitopsEngineCluster. So I have added this boilerplate.

Reply:
@Rizwana777 I would say it's fine for us to have 2 GitOpsEngineInstances created by the test, as long as having 2 doesn't break the functionality of the test itself. I'd suggest trying out using sample data, and seeing if it works (presuming it makes sense).

Reply:
@jgwest  used `db. CreateSampleData()` wherever possible.